libres/lib/res_util/matrix.cpp

```diff
@@ -1128,122 +302,19 @@ void matrix_set_row(matrix_type *matrix, const double *data, int row) {
 
-/*
-   This function will return the double data pointer of the matrix,
-   when you use this explicitly you ARE ON YOUR OWN.
-*/
-
-double *matrix_get_data(const matrix_type *matrix) { return matrix->data; }
-
-/*
-   The query functions below can be used to ask for the dimensions &
-   strides of the matrix.
-*/
-
-int matrix_get_rows(const matrix_type *matrix) { return matrix->rows; }
-
-int matrix_get_columns(const matrix_type *matrix) { return matrix->columns; }
-
-int matrix_get_column_stride(const matrix_type *matrix) {
-    return matrix->column_stride;
-}
-
-void matrix_get_dims(const matrix_type *matrix, int *rows, int *columns,
-                     int *row_stride, int *column_stride) {
-
-    *rows = matrix->rows;
-    *columns = matrix->columns;
-    *row_stride = matrix->row_stride;
-    *column_stride = matrix->column_stride;
-}
-
-bool matrix_is_quadratic(const matrix_type *matrix) {
-    if (matrix->rows == matrix->columns)
-        return true;
-    else
-        return false;
-}
-
-/*
-   Goes through all the elements in the matrix - and return true if they are all finite.
-*/
-
-#ifdef ERT_HAVE_ISFINITE
-
-bool matrix_is_finite(const matrix_type *matrix) {
-    int i, j;
-    for (i = 0; i < matrix->rows; i++)
-        for (j = 0; j < matrix->columns; j++)
-            if (!std::isfinite(matrix->data[GET_INDEX(matrix, i, j)])) {
-                printf("%s(%d,%d) = %g \n", matrix->name, i, j,
-                       matrix->data[GET_INDEX(matrix, i, j)]);
-                return false;
-            }
-
-    return true;
-}
-
-void matrix_assert_finite(const matrix_type *matrix) {
-    if (!matrix_is_finite(matrix)) {
-        if ((matrix->rows * matrix->columns) < 400)
-            matrix_pretty_fprint(matrix, matrix->name, " %6.3f", stdout);
-
-        util_abort("%s: matrix:%s is not finite. \n", __func__, matrix->name);
-    }
-}
-
-#endif
-
-/*
-   Return true if the two matrices m1 and m2 are equal. The equality
-   test is based on element-by-element memcmp() comparison, i.e. the
-   there is ZERO numerical tolerance in the comparison.
-
-   If the two matrices do not have equal dimension false is returned.
-*/
+int matrix_get_columns(const matrix_type *matrix) { return matrix->cols(); }
 
 bool matrix_equal(const matrix_type *m1, const matrix_type *m2) {
-    if (!((m1->rows == m2->rows) && (m1->columns == m2->columns)))
-        return false;
-    {
-        int i, j;
-        for (i = 0; i < m1->rows; i++) {
-            for (j = 0; j < m1->columns; j++) {
-                int index1 = GET_INDEX(m1, i, j);
-                int index2 = GET_INDEX(m2, i, j);
-                double d1 = m1->data[index1];
-                double d2 = m2->data[index2];
-
-                if (d1 != d2)
-                    return false;
-            }
-        }
-    }
-
-    /* OK - we came all the way through - they are equal. */
-    return true;
+    // return *m1 == *m2;
+    return matrix_similar(m1, m2, 1e-5);
```

Comment:
Decreas epsilon? Now it is identical to all the other calls to `matrix_similar`, so might as well replace is outright if we keep 1e-5 as epsilon.

Reply:
I forgot to uncomment the old code. This should be `*m1 == *m2`.